File: XUnitTest/__mock__/MockMusic.cs
using Application.Account.Dto;
using Bogus;
using Domain.Streaming.Agreggates;
using Domain.Streaming.ValueObject;

namespace __mock__;
public class MockMusic
{
    private static readonly Lazy<MockMusic> _instance = new Lazy<MockMusic>(() => new MockMusic());

    private MockMusic() { }

    public static MockMusic Instance => _instance.Value;

    public Music GetFaker()
    {
        var fakeMusic = new Faker<Music>()
            .RuleFor(m => m.Id, f => f.Random.Guid())
            .RuleFor(m => m.Name, f => f.Commerce.ProductName())
            .RuleFor(m => m.Duration, f => new Duration(f.Random.Int(60, 600)))
            .RuleFor(m => m.Album, new Album())
            .RuleFor(m => m.Playlists, new List<Playlist>())
            .Generate();
        
        return fakeMusic;
    }

    public List<Music> GetListFaker(int count)
    {
        var mockPlaylist = MockPlaylist.Instance.GetFaker();
        var musicList = new List<Music>();
        for (var i = 0; i < count; i++)
        {
            var mockMusic = GetFaker();
            mockPlaylist.Musics.Add(mockMusic);
            mockMusic.Playlists.Add(mockPlaylist);
            musicList.Add(mockMusic);
        }
        return musicList;
    }

    public MusicDto GetDtoFromMusic(Music music)
    {
        var musicDto = new MusicDto
        {
            Id = music.Id,
            Name = music.Name,
            Duration = music.Duration.Value,
            FlatId = Guid.NewGuid()
        };

        return musicDto;
    }

    public List<MusicDto> GetDtoListFromMusicList(IList<Music> musics)
    {
        var musicDtoList = new List<MusicDto>();
        foreach (var music in musics)
        {
            var musicDto = GetDtoFromMusic(music);
            musicDtoList.Add(musicDto);
        }
        return musicDtoList;
    }
}

File: AngularApp/src/app/__mocks__/mock.music.ts
import { Music } from '../model';
import { faker } from '@faker-js/faker';

export class MockMusic {
  private static _instance: MockMusic;
  private faker: typeof faker;

  private constructor() {
    this.faker = faker;
  }

  public static instance(): MockMusic {
    if (!MockMusic._instance) {
      MockMusic._instance = new MockMusic();
    }
    return MockMusic._instance;
  }

  public getFaker(): Music {
    const mockMusic: Music = ({
      id: faker.string.uuid(),
      name: faker.music.songName(),
      duration: faker.number.int(),
      url: faker.internet.url(),
    });
    return mockMusic;
  }

  public generateMusicList(count: number): Music[] {
    const MusicList: Music[] = [];
    for (let i = 0; i < count; i++) {
      const Music: Music = this.getFaker();
      MusicList.push(Music);
    }
    return MusicList;
  }
}
